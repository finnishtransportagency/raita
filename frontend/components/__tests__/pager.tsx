import * as F from 'fast-check';
import { fireEvent, logDOM, logRoles, render } from '@testing-library/react';

import Pager from '../pager';

describe('Pager', () => {
  test('base implementation', () => {
    const r = render(<Pager count={50} size={5} page={0} />);

    expect(r).toMatchSnapshot();
  });
});
