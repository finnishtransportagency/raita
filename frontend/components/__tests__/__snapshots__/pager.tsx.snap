// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pager base implementation 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        aria-label="common:pager_a11y_label"
        class="root"
        role="navigation"
      >
        <ul
          class="flex items-center space-x-2 pager"
        >
          <li
            class="page pageCurrent"
          >
            <button
              aria-current="true"
              aria-label="common:pager_a11y_goto_page"
              class="pageButton"
            >
              1
            </button>
          </li>
          <li
            class="page"
          >
            <button
              aria-label="common:pager_a11y_goto_page"
              class="pageButton"
            >
              2
            </button>
          </li>
          <li
            class="page"
          >
            <button
              aria-label="common:pager_a11y_goto_page"
              class="pageButton"
            >
              3
            </button>
          </li>
          <li
            class="page"
          >
            <button
              aria-label="common:pager_a11y_goto_page"
              class="pageButton"
            >
              4
            </button>
          </li>
          <li
            class="page"
          >
            <button
              aria-label="common:pager_a11y_goto_page"
              class="pageButton"
            >
              5
            </button>
          </li>
          <li
            class="page"
          >
            <button
              aria-label="common:pager_a11y_goto_page"
              class="pageButton"
            >
              6
            </button>
          </li>
          <li
            class="page"
          >
            <button
              aria-label="common:pager_a11y_goto_page"
              class="pageButton"
            >
              7
            </button>
          </li>
          <li
            class="page"
          >
            <button
              aria-label="common:pager_a11y_goto_page"
              class="pageButton"
            >
              8
            </button>
          </li>
          <li
            class="page"
          >
            <button
              aria-label="common:pager_a11y_goto_page"
              class="pageButton"
            >
              9
            </button>
          </li>
          <li
            class="page"
          >
            <button
              aria-label="common:pager_a11y_goto_page"
              class="pageButton"
            >
              10
            </button>
          </li>
        </ul>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      aria-label="common:pager_a11y_label"
      class="root"
      role="navigation"
    >
      <ul
        class="flex items-center space-x-2 pager"
      >
        <li
          class="page pageCurrent"
        >
          <button
            aria-current="true"
            aria-label="common:pager_a11y_goto_page"
            class="pageButton"
          >
            1
          </button>
        </li>
        <li
          class="page"
        >
          <button
            aria-label="common:pager_a11y_goto_page"
            class="pageButton"
          >
            2
          </button>
        </li>
        <li
          class="page"
        >
          <button
            aria-label="common:pager_a11y_goto_page"
            class="pageButton"
          >
            3
          </button>
        </li>
        <li
          class="page"
        >
          <button
            aria-label="common:pager_a11y_goto_page"
            class="pageButton"
          >
            4
          </button>
        </li>
        <li
          class="page"
        >
          <button
            aria-label="common:pager_a11y_goto_page"
            class="pageButton"
          >
            5
          </button>
        </li>
        <li
          class="page"
        >
          <button
            aria-label="common:pager_a11y_goto_page"
            class="pageButton"
          >
            6
          </button>
        </li>
        <li
          class="page"
        >
          <button
            aria-label="common:pager_a11y_goto_page"
            class="pageButton"
          >
            7
          </button>
        </li>
        <li
          class="page"
        >
          <button
            aria-label="common:pager_a11y_goto_page"
            class="pageButton"
          >
            8
          </button>
        </li>
        <li
          class="page"
        >
          <button
            aria-label="common:pager_a11y_goto_page"
            class="pageButton"
          >
            9
          </button>
        </li>
        <li
          class="page"
        >
          <button
            aria-label="common:pager_a11y_goto_page"
            class="pageButton"
          >
            10
          </button>
        </li>
      </ul>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
